index.js

```diff
@@ -112,10 +126,18 @@ module.exports = function(gitPath) {
 
       // Find branch and SHA
       if (refPath) {
-        var branchPath = path.join(gitPath, refPath.trim());
+        refPath = refPath.trim();
+        var branchPath = path.join(gitPath, refPath);
 
         result.branch  = branchName;
-        result.sha     = fs.readFileSync(branchPath, {encoding: 'utf8' }).trim();
+        try {
```

Comment:
I'd prefer to use `exists-sync` package to detect if the file exists (versus `try` / `catch`).